plugins/tabletools/plugin.js

```diff
@@ -325,27 +330,34 @@
 					cell = new CKEDITOR.dom.element( mapRow[ i ] );
 
 				if ( cell.$ ) {
-					if ( cell.$.colSpan == 1 )
+					if ( cell.$.colSpan == 1 ) {
 						cell.remove();
-					// Reduce the col spans.
-					else
+					} else {
+						// Reduce the col spans.
 						cell.$.colSpan -= 1;
+					}
 
 					j += cell.$.rowSpan - 1;
 
-					if ( !row.$.cells.length )
+					if ( !row.$.cells.length ) {
 						rowsToDelete.push( row );
+					}
 				}
 			}
 		}
 
-		var firstRowCells = table.$.rows[ 0 ] && table.$.rows[ 0 ].cells;
-
 		// Where to put the cursor after columns been deleted?
 		// 1. Into next cell of the first row if any;
 		// 2. Into previous cell of the first row if any;
 		// 3. Into table's parent element;
-		var cursorPosition = new CKEDITOR.dom.element( firstRowCells[ startColIndex ] || ( startColIndex ? firstRowCells[ startColIndex - 1 ] : table.$.parentNode ) );
+		var cursorPosition;
+		if ( map[ 0 ].length - 1 > endColIndex && map[ 0 ][ endColIndex + 1 ].cellIndex !== -1 ) {
```

Comment:
First part of condition `map[ 0 ].length - 1 > endColIndex` check if there was some columns on the right after deleted part.
- `map[ 0 ].length - 1` is the last index of possible column value
- `endColIndex` is last index of deleted column.

Second part of condition `map[ 0 ][ endColIndex + 1 ].cellIndex !== -1` checks, if cell (`endColIndex + 1`) after deleted column, where we want to put cursor, is still in table. Theoretically it should be true, if first part is true ;)